crates/store/re_entity_db/src/entity_db.rs

```diff
@@ -466,35 +466,34 @@ impl EntityDb {
             .store_info_msg()
             .map(|msg| Ok(LogMsg::SetStoreInfo(msg.clone())));
 
-        let time_filter = time_selection.map(|(timeline, range)| {
-            (
-                timeline,
-                ResolvedTimeRange::new(range.min.floor(), range.max.ceil()),
-            )
-        });
+        let data_messages = {
+            let mut chunks: Vec<&Arc<Chunk>> = self.store().iter_chunks().collect();
+
+            if let Some((timeline, range)) = time_selection {
+                let time_range = ResolvedTimeRange::new(range.min.floor(), range.max.ceil());
+                chunks.retain(|chunk| {
+                    // TODO(cmc): chunk.slice_time_selection(time_selection)
+                    chunk
+                        .timelines()
+                        .get(&timeline)
+                        .map_or(false, |time_column| {
+                            time_range.contains(time_column.time_range().min())
+                                || time_range.contains(time_column.time_range().max())
+                        })
+                });
+            }
 
-        let data_messages = self
-            .store()
-            .iter_chunks()
-            .filter(move |chunk| {
-                let Some((timeline, time_range)) = time_filter else {
-                    return true;
-                };
+            // Try to roughly preserve the order of the chunks
+            // from how they were originally logged.
+            // See https://github.com/rerun-io/rerun/issues/7175 for why.
+            chunks.sort_by_key(|chunk| chunk.row_ids().next());
```

Comment:
In practice as of today this is fine, but `row_ids().next()`'s ordering is undefined really.
```suggestion
            chunks.sort_by_key(|chunk| chunk.row_id_range().map(|(min, _)| min);

Reply:
undefined how?

Reply:
Ah, they can be unsorted :+1:

Reply:
The chunk might not be sorted, `row_id_range` makes sure to handle that case.